lib_bgp_data/blacklist/blacklist_parser.py

```diff
@@ -0,0 +1,110 @@
+        """
+        sources = {'uce2': 'http://wget-mirrors.uceprotect.net/rbldnsd-all/dnsbl-2.uceprotect.net.gz', 'uce3': 'http://wget-mirrors.uceprotect.net/rbldnsd-all/dnsbl-3.uceprotect.net.gz', 'spamhaus': 'https://www.spamhaus.org/drop/asndrop.txt', 'mit': 'https://raw.githubusercontent.com/ctestart/BGP-SerialHijackers/master/prediction_set_with_class.csv'}
+        output_str = dict()
+        for typ in sources.keys():
+            output_str[typ] = ''
+        # For each source in the sources dict, GET url and save string
+        # to dict.
+        for source in sources.keys():
+            downloaded_file = requests.get(sources[source])
+            # UCEPROTECT is inconsistent in that sometimes a python
+            # requested file will sometimes come as a txt, sometimes
+            # as a tarball that needs to be unzipped. We check this:
+            if downloaded_file.headers['Content-Type'] == 'application/octet-stream':
+                decompressed = gzip.decompress(downloaded_file.content)
+                # UCEPROTECT uses ISO-8859-1 for text, not utf-8
+                output_str[source] = decompressed.decode('ISO-8859-1')
+            else:
+                # We have a txt and save to dict directly
+                output_str[source] = downloaded_file.text
+        return output_str
+
+    def _parse_lists(self, outputs: dict):
+        """For each source in outputs, takes the string attached to
+        the source and parses it for ASNs, then saves to dict with
+        sources as keys with a list of ASNs
+        """
+        parsed = dict()
+        # For each source in outputs, parse for ASNs and save ASNs as
+        # list for each source in dict.
+        for output in outputs.keys():
+            # If not a mit csv, just regex to find ASNs.
+            if output != 'mit':
+                parsed[output] = re.findall(r'AS\d+', outputs[output])
+            # If mit csv, use csv utilities to ID malicious ASNs.
+            if output == 'mit':
+                parsed[output] = []
+                fake_file = StringIO(outputs[output])
+                mit_reader = csv.DictReader(fake_file, delimiter=',')
+                for row in mit_reader:
+                    if row['HardVotePred'] == '1':
+                        parsed[output].append(row['ASN'])
+        # For each key in parsed dictionary, remove duplicates and
+        # remove 'AS' from infront of ASN.
+        for key in parsed.keys():
+            parsed[key] = list(dict.fromkeys(parsed[key]))
+            # Remove AS from the the numbers
+            for i in range(len(parsed[key])):
+                parsed[key][i] = parsed[key][i].replace('AS', '')
+        return parsed
+
+    def _format_dict(self, parsed: dict):
+        """Takes a dict, with the header as the keys, and converts into
+        a formatted list for input into database"""
+        formatted = []
+        for key in parsed.keys():
```

Comment:
return [v, k for k, v in parsed.items()]? I'm not sure this needs it's own function